File: Java/BinarySearch/First Bad Version.java
//https://leetcode.com/problems/first-bad-version/
public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int l = 1, h = n;
        int ans = -1;
        while (l <= h) {
            int mid = l + (h - l) / 2;
            if (isBadVersion(mid)) {
                ans = mid;
                h = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        return ans;
    }
}

File: Python/BinarySearch/First Bad Version.py
//https://leetcode.com/problems/first-bad-version/
class Solution:
    def firstBadVersion(self, n: int) -> int:
        l, h = 1, n
        ans = -1
        while l <= h:
            mid = l + (h - l) // 2
            if isBadVersion(mid):
                ans = mid
                h = mid - 1
            else:
                l = mid + 1
        return ans
